lib/matplotlib/cbook/__init__.py

```diff
@@ -2156,3 +2156,64 @@ def _check_in_list(values, **kwargs):
             raise ValueError(
                 "{!r} is not a valid value for {}; supported values are {}"
                 .format(v, k, ', '.join(map(repr, values))))
+
+
+_valid_compass = ['NE', 'NW', 'SW', 'SE', 'E', 'W', 'E', 'S', 'N', 'C']
+
+
+def _map_loc_to_compass(loc, allowtuple=False, allowbest=False,
+                        fallback='NE', warnonly=True):
```

Comment:
Not too important since it's a private API, but I would set the default `warnonly=False`. We actually want to fail for invalid input, and do only have `warnonly` to support a deprecation for default fallback.

With `warnonly=False`, we can already leave out the parameter in cases which are not affected by the deprecation and thus have simpler code there.